src/test/resources/com/puppycrawl/tools/checkstyle/checks/design/finalclass/InputFinalClassInnerAndNestedClass.java

```diff
@@ -0,0 +1,76 @@
+/*
+FinalClass
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.design.finalclass;
+
+public class InputFinalClassInnerAndNestedClass {
+
+    private class SuperClass { // ok
+        private SuperClass() {
+        }
+    }
+
+    private class SubClass extends SuperClass {
+    }
+
+    class SameName { // violation
+        private SameName() {
+        }
+    }
+
+    static class Test {
+        static class SameName { // ok
+            private SameName() {
+            }
+            class Test3 {
+            }
+        }
+    }
+
+    class TestInnerClass {
+        class SameName { // ok
+            class Test3 {
+                class Test4 extends SameName {
+                }
+            }
+            private SameName() {
+            }
+        }
+    }
+
+    class TestNestedClasses {
+        class SameName { // violation
+            private SameName() {
+            }
+            class Test3 {
+                class Test4 extends Test.SameName {
+                }
+            }
+        }
+    }
+}
+
+enum foo {
+    VALUE_1, VALUE_2;
+
+    class A {
+        class B { // ok
+            private B() {
+            }
+        }
+
+        class c extends B {
+        }
+
+        class D {
+            class B { // violation
+                private B() {
+                }
+            }
+        }
+    }
```

Comment:
Please add
```java

        class E {
            class B { // violation
                private B() {
                }
            }
        }
```
and this will solve your pitest problem (also, you need two `break`s, see my other comment)

Reply:
Adding `break` would break the logic.